builtin/game/item_entity.lua

```diff
@@ -152,6 +175,74 @@ core.register_entity(":__builtin:item", {
 			return
 		end
 
+		local is_stuck = false
+		local snode = core.get_node_or_nil(pos)
+		if snode then
+			local sdef = core.registered_nodes[snode.name] or {}
+			is_stuck = (sdef.walkable == nil or sdef.walkable == true)
+				and (sdef.collision_box == nil or sdef.collision_box.type == "regular")
+				and (sdef.node_box == nil or sdef.node_box.type == "regular")
+		end
+
+		-- Push item out when stuck inside solid node
+		if is_stuck then
+			local shootdir
+			local order = {
+				{x=1, y=0, z=0}, {x=-1, y=0, z= 0},
+				{x=0, y=0, z=1}, {x= 0, y=0, z=-1},
+			}
+
+			-- Check which one of the 4 sides is free
+			for o = 1, #order do
+				local cnode = core.get_node(vector.add(pos, order[o])).name
+				local cdef = core.registered_nodes[cnode] or {}
+				if cnode ~= "ignore" and cdef.walkable == false then
+					shootdir = order[o]
+					break
+				end
+			end
+			-- If none of the 4 sides is free, check upwards
```

Comment:
The node below could be free.

Reply:
Intentionally not checked.

Reply:
But why?
The item could simply fall out.
Is it to make item elevators easier?

Reply:
The only way this could happen is if someone places a node where the item is right while it's falling down, should be quite rare so it's a good choice not to care about that.
>Is it to make item elevators easier?

This is a nice side effect, also.